### ISSUE TITLE
0.9.28 - After upgrade, crashes

### ISSUE THREAD

**Issue by vikingtoby** [jhuckaby/Cronicle#625]
## Summary

After updating to 0.9.28 from 0.9.27, the cronicle service keeps crashing on my single master setup, worker nodes are fine.

## Steps to reproduce the problem

Update to 0.9.28, start service, wait 3-4 min.

Infinitely reproducible on my setup, if I start the service again, it crashes immediately.

## Your Setup

1x master
3x workers

### Operating system and version?

AlmaLinux release 8.8 (Sapphire Caracal)

### Node.js version?

nodejs-full-i18n-16.20.1-1.module_el8.8.0+3594+e74fe53f.x86_64
nodejs-16.20.1-1.module_el8.8.0+3594+e74fe53f.x86_64
nodejs-docs-16.20.1-1.module_el8.8.0+3594+e74fe53f.noarch

### Cronicle software version?

0.9.28

### Are you using a multi-server setup, or just a single server?

Multi-server, but single master

### Are you using the filesystem as back-end storage, or S3/Couchbase?

ZFS, local filesystem

### Can you reproduce the crash consistently?

Yes.

## Log Excerpts

```
[1692379662.246][2023-08-18 19:27:42][<redacted hostname>][2701][Cronicle][debug][1][Uncaught Exception: RangeError [ERR_OUT_OF_RANGE]: The value of "end" is out of range. It must be >= 0 && <= 9007199254740991. Received -1][RangeError [ERR_OUT_OF_RANGE]: The value of "end" is out of range. It must be >= 0 && <= 9007199254740991. Received -1     at new ReadStream (node:internal/fs/streams:205:5)     at Object.createReadStream (node:fs:2916:10)     at /opt/cronicle/node_modules/pixl-server-web/lib/static.js:153:20     at /opt/cronicle/node_modules/pixl-server-web/node_modules/async/dist/async.js:3648:19     at wrapper (/opt/cronicle/node_modules/pixl-server-web/node_modules/async/dist/async.js:271:20)     at replenish (/opt/cronicle/node_modules/pixl-server-web/node_modules/async/dist/async.js:441:29)     at iterateeCallback (/opt/cronicle/node_modules/pixl-server-web/node_modules/async/dist/async.js:430:21)     at /opt/cronicle/node_modules/pixl-server-web/node_modules/async/dist/async.js:327:20     at /opt/cronicle/node_modules/pixl-server-web/node_modules/async/dist/async.js:3646:17     at processTicksAndRejections (node:internal/process/task_queues:78:11)]
[1692379662.246][2023-08-18 19:27:42][<redacted hostname>][2701][Cronicle][debug][1][Emergency Shutdown: RangeError [ERR_OUT_OF_RANGE]: The value of "end" is out of range. It must be >= 0 && <= 9007199254740991. Received -1][]
```

**Comment by jhuckaby** [comment]
This crash is happening inside the web server (pixl-server-web) which is included with Cronicle.  It looks like it's receiving some kind of rogue HTTP request with an invalid `Range` header.

Is your server publicly accessible on the internet by any chance?  This could be some kind of hacking attempt by a bot.

I will try to track down this issue in the web server and patch it ASAP.

Can you please send me your `/opt/cronicle/logs/WebServer.log` file?  My email is `jhuckaby AT gmail DOT com`.  This will help me track down the issue.

Thank you.

**Comment by vikingtoby** [comment]
Nope, only accessible internally.
But we heavily use the API to manage our events, perhaps that's the trigger?

**Comment by jhuckaby** [comment]
Okay, I see how this could possibly happen, even without a bad `Range` header.  If you are trying to view a 0-byte file via HTTP (perhaps a 0-byte job log file?), I can see how the code could crash.  This line here:

https://github.com/jhuckaby/pixl-server-web/blob/master/lib/static.js#L122

I'll add some protection in here and release a patch.

**Comment by vikingtoby** [comment]
Awesome, thanks!

I triggered it 5x times after the first upgrade, 3-4 min every time.

This second upgrade, I can't seem to reproduce, wonder what happened.
My backup is some hours old, perhaps the job that created an empty log like you mentioned, and that job hasn't run today, or hasn't produced a file like that?

I'll keep an eye on it and keep it on 0.9.28 for now (until you release the new patch), and see what happens.

**Comment by jhuckaby** [comment]
Published [v0.9.29](https://github.com/jhuckaby/Cronicle/releases/tag/v0.9.29) with a fix.  Let me know if it happens again 😊 

Thanks!
